### PR TITLE
check if forwarding is enabled

### PR DESCRIPTION
**Pull request by ferraith**

### CODE REVIEW

File: java/com/engflow/notificationqueue/Client.java [hunk 1]
```diff
@@ -89,10 +90,14 @@ public static void main(String[] args) throws Exception {
     } finally {
       if (channel != null) {
         channel.shutdownNow();
-        forwardChannel.shutdownNow();
+        if (shouldForwared) {
```
Comment (Yannic) [inline java/com/engflow/notificationqueue/Client.java:93]: Maybe check whether forwardChannel is not null here instead?

File: java/com/engflow/notificationqueue/Client.java [hunk 2]
```diff
@@ -89,10 +90,14 @@ public static void main(String[] args) throws Exception {
...
+          forwardChannel.shutdownNow();
+        }
         try {
           channel.awaitTermination(10, TimeUnit.SECONDS);
-          forwardChannel.awaitTermination(10, TimeUnit.SECONDS);
+          if (shouldForwared) {
```
Comment (Yannic) [inline java/com/engflow/notificationqueue/Client.java:98]: dito

**Review by Yannic — COMMENTED** [review]
Thanks a lot for sending this!

File: java/com/engflow/notificationqueue/Client.java (hunk 1 above)
Comment (ferraith) [reply java/com/engflow/notificationqueue/Client.java]: Fixed!

File: java/com/engflow/notificationqueue/Client.java (hunk 2 above)
Comment (ferraith) [reply java/com/engflow/notificationqueue/Client.java]: Fixed!

**Review by Yannic — APPROVED** [review]

**Pull request merged**